runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -528,8 +528,13 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       J9ROMClass *romClass = NULL;
       if (!(romClass = JITServerHelpers::getRemoteROMClassIfCached(clientSession, clazz)))
          {
-         romClass = JITServerHelpers::romClassFromString(std::get<0>(classInfoTuple), compInfo->persistentMemory());
-         JITServerHelpers::cacheRemoteROMClass(getClientData(), clazz, romClass, &classInfoTuple);
+         //check whether the first argument of the classInfoTuple is an empty string
```

Comment:
The the ROMClass is not already cached at the server we get the information from the request sent by the client and cache it. If the client did not send us anything, the server needs to specifically ask about the information, because it cannot continue with the ROMClas.